### PR TITLE
BGM and voice file replacement

### PR DESCRIPTION
**Pull request by jonko0493**
Adds Extract, Replace, and Restore buttons to BGM & Voiced Line editors. They're all functional with fully-functional ADX/AHX encoding.

### CODE REVIEW

**Comment by jonko0493** [comment]
I know what's wrong with the tests. Will fix in the morning lol.

**Comment by jonko0493** [comment]
jk didn't sleep as early as i thought so i just fixed the tests now lmao

File: src/SerialLoops/Editors/BackgroundMusicEditor.cs [hunk 1]
```diff
@@ -20,9 +23,54 @@ public override Container GetEditorPanel()
...
+                    }, tracker, "Replace BGM track");
+                }
+            };
+
+            Button restoreButton = new() { Text = "Restore" };
+            restoreButton.Click += (obj, args) =>
+            {
+                File.Copy(Path.Combine(_project.BaseDirectory, "original", "bgm", Path.GetFileName(_bgm.BgmFile)), Path.Combine(_project.BaseDirectory, _bgm.BgmFile), true);
+                File.Copy(Path.Combine(_project.IterativeDirectory, "original", "bgm", Path.GetFileName(_bgm.BgmFile)), Path.Combine(_project.IterativeDirectory, _bgm.BgmFile), true);
+                Content = GetEditorPanel();
```
Comment (WiIIiam278) [inline src/SerialLoops/Editors/BackgroundMusicEditor.cs:57]: One slight issue with this is that it doesn't kill playing tracks -- maybe invoke the stop method on the player panel before calling this? Same with both replace and restore.

**Review by WiIIiam278 — APPROVED** [review]
Really fantastic, works with both audio and BGM -- only one minor snag in my testing

**Pull request merged**